Nextjs import / webpack

It (0.1.5) seems to fail with webpack 5 / next js 15.3.4.

I really have no idea why after diving into it a bit.  Tested it on a vanilla install.

3x
<w> [webpack.cache.PackFileCacheStrategy] Serializing big strings (114kiB) impacts deserialization performance (consider using Buffer instead and decode when needed)
Failed to compile.

./node_modules/@sparkjsdev/spark/dist/spark.module.js
Module not found: Invalid generator object. Asset Modules Plugin has been initialized using a generator object that does not match the API schema.
 - generator has an unknown property 'filename'. These properties are valid:     
   object { binary?, dataUrl? }
   -> Generator options for asset/inline modules.

--- a/vite.config.ts
+++ b/vite.config.ts
@@ -1,8 +1,35 @@
 import fs from "node:fs";
 import path from "node:path";
+import type { Plugin } from "vite";
 import { defineConfig } from "vite";
 import dts from "vite-plugin-dts";
 import glsl from "vite-plugin-glsl";
+
+/**
+ * Vite plugin to fix WASM data URL compatibility with webpack/Next.js.
+ *
+ * wasm-pack generates code like: new URL("data:...", import.meta.url)
+ * The import.meta.url argument is unnecessary for data: URLs and causes
+ * webpack/Vite to incorrectly try to rewrite the URL as a file path.
+ *
+ * This plugin transforms:
+ *   new URL("data:...", import.meta.url) → new URL("data:...")
+ *
+ * See: https://github.com/sparkjsdev/spark/issues/95
+ */
+function fixWasmDataUrl(): Plugin {
+  return {
+    name: "fix-wasm-data-url",
+    renderChunk(code) {
+      // Match: new URL("data:...", import.meta.url)
+      // The data URL can contain any characters including quotes (escaped)
+      const dataUrlPattern =
+        /new\s+URL\(\s*("data:[^"]*")\s*,\s*import\.meta\.url\s*\)/g;
+      const result = code.replace(dataUrlPattern, "new URL($1)");
+      return result !== code ? result : null;
+    },
+  };
+}
 
 const assetsDirectory = "examples/assets";
 const localAssetsDirectoryExist = fs.existsSync(assetsDirectory);
@@ -32,6 +59,9 @@ export default defineConfig(({ mode }) => {
       }),
 
       dts({ outDir: "dist/types" }),
+
+      // Fix webpack/Next.js compatibility for WASM data URLs
+      fixWasmDataUrl(),
       {
         name: "serve-node-modules-alias",
         configureServer(server) {

